### ISSUE TITLE
npm ERR! typeerror Error: Missing required argument #1 getting this error while npm install on server

### ISSUE THREAD

**Issue by Prudhvi1357** [npm/npm#12766]
npm ERR! typeerror Error: Missing required argument #1 
npm ERR! typeerror     at markDeps (/usr/local/lib/node_modules/npm/lib/install/deps.js:106:5) 
npm ERR! typeerror     at /usr/local/lib/node_modules/npm/node_modules/slide/lib/async-map.js:52:35 npm ERR! typeerror     at Array.forEach (native) 
npm ERR! typeerror     at /usr/local/lib/node_modules/npm/node_modules/slide/lib/async-map.js:52:11 npm ERR! typeerror     at Array.forEach (native) 
npm ERR! typeerror     at asyncMap (/usr/local/lib/node_modules/npm/node_modules/slide/lib/async-map.js:51:8) 
npm ERR! typeerror     at Array.0 (/usr/local/lib/node_modules/npm/node_modules/slide/lib/bind-actor.js:15:8) 
npm ERR! typeerror     at LOOP (/usr/local/lib/node_modules/npm/node_modules/slide/lib/chain.js:15:14) 
npm ERR! typeerror     at chain (/usr/local/lib/node_modules/npm/node_modules/slide/lib/chain.js:20:5) npm ERR! typeerror     at recalculateMetadata (/usr/local/lib/node_modules/npm/lib/install/deps.js:147:3)

**Comment by afraser** [comment]
Same:

```
npm install
npm WARN deprecated graceful-fs@2.0.3: graceful-fs v3.0.0 and before will fail on node releases >= v7.0. Please update to graceful-fs@^4.0.0 as soon as possible. Use 'npm ls graceful-fs' to find it in the tree.
npm ERR! Linux 3.13.0-86-generic
npm ERR! argv "/home/ubuntu/nvm/versions/node/v5.3.0/bin/node" "/home/ubuntu/nvm/versions/node/v5.3.0/bin/npm" "install"
npm ERR! node v5.3.0
npm ERR! npm  v3.9.0
npm ERR! code EMISSINGARG

npm ERR! typeerror Error: Missing required argument #1
npm ERR! typeerror     at markDeps (/home/ubuntu/nvm/versions/node/v5.3.0/lib/node_modules/npm/lib/install/deps.js:106:5)
npm ERR! typeerror     at /home/ubuntu/nvm/versions/node/v5.3.0/lib/node_modules/npm/node_modules/slide/lib/async-map.js:52:35
npm ERR! typeerror     at Array.forEach (native)
npm ERR! typeerror     at /home/ubuntu/nvm/versions/node/v5.3.0/lib/node_modules/npm/node_modules/slide/lib/async-map.js:52:11
npm ERR! typeerror     at Array.forEach (native)
npm ERR! typeerror     at asyncMap (/home/ubuntu/nvm/versions/node/v5.3.0/lib/node_modules/npm/node_modules/slide/lib/async-map.js:51:8)
npm ERR! typeerror     at Array.<anonymous> (/home/ubuntu/nvm/versions/node/v5.3.0/lib/node_modules/npm/node_modules/slide/lib/bind-actor.js:15:8)
npm ERR! typeerror     at LOOP (/home/ubuntu/nvm/versions/node/v5.3.0/lib/node_modules/npm/node_modules/slide/lib/chain.js:15:14)
npm ERR! typeerror     at chain (/home/ubuntu/nvm/versions/node/v5.3.0/lib/node_modules/npm/node_modules/slide/lib/chain.js:20:5)
npm ERR! typeerror     at recalculateMetadata (/home/ubuntu/nvm/versions/node/v5.3.0/lib/node_modules/npm/lib/install/deps.js:147:3)
npm ERR! typeerror This is an error with npm itself. Please report this error at:
npm ERR! typeerror     <http://github.com/npm/npm/issues>

npm ERR! Please include the following file with any support request:
npm ERR!     /home/ubuntu/learning/npm-debug.log

npm install returned exit code 1

Action failed: npm install
```

**Comment by afraser** [comment]
Was using npm@3.9.0 (with node v5.3.0).
Fixed by rolling back to npm@3.8.9.

**Comment by ffxsam** [comment]
Broken, big time. Same here. npm-debug.log follows.

```
0 info it worked if it ends with ok
1 verbose cli [ '/Users/samh/.nvm/versions/node/v6.2.0/bin/node',
1 verbose cli   '/Users/samh/.nvm/versions/node/v6.2.0/bin/npm',
1 verbose cli   'update' ]
2 info using npm@3.8.9
3 info using node@v6.2.0
4 verbose stack Error: Missing required argument #1
4 verbose stack     at module.exports (/Users/samh/.nvm/versions/node/v6.2.0/lib/node_modules/npm/lib/install/mutate-into-logical-tree.js:12:3)
4 verbose stack     at /Users/samh/.nvm/versions/node/v6.2.0/lib/node_modules/npm/lib/outdated.js:78:5
4 verbose stack     at /Users/samh/.nvm/versions/node/v6.2.0/lib/node_modules/npm/lib/outdated.js:62:20
4 verbose stack     at /Users/samh/.nvm/versions/node/v6.2.0/lib/node_modules/npm/node_modules/read-package-tree/rpt.js:186:9
4 verbose stack     at f (/Users/samh/.nvm/versions/node/v6.2.0/lib/node_modules/npm/node_modules/once/once.js:17:25)
4 verbose stack     at /Users/samh/.nvm/versions/node/v6.2.0/lib/node_modules/npm/node_modules/read-package-tree/rpt.js:152:20
4 verbose stack     at f (/Users/samh/.nvm/versions/node/v6.2.0/lib/node_modules/npm/node_modules/once/once.js:17:25)
4 verbose stack     at then (/Users/samh/.nvm/versions/node/v6.2.0/lib/node_modules/npm/node_modules/read-package-tree/rpt.js:121:22)
4 verbose stack     at /Users/samh/.nvm/versions/node/v6.2.0/lib/node_modules/npm/node_modules/read-package-tree/rpt.js:81:20
4 verbose stack     at FSReqWrap.oncomplete (fs.js:117:15)
5 verbose cwd /Volumes/SuperData/Sites/harris/L11
6 error Darwin 15.5.0
7 error argv "/Users/samh/.nvm/versions/node/v6.2.0/bin/node" "/Users/samh/.nvm/versions/node/v6.2.0/bin/npm" "update"
8 error node v6.2.0
9 error npm  v3.8.9
10 error code EMISSINGARG
11 error typeerror Error: Missing required argument #1
11 error typeerror     at module.exports (/Users/samh/.nvm/versions/node/v6.2.0/lib/node_modules/npm/lib/install/mutate-into-logical-tree.js:12:3)
11 error typeerror     at /Users/samh/.nvm/versions/node/v6.2.0/lib/node_modules/npm/lib/outdated.js:78:5
11 error typeerror     at /Users/samh/.nvm/versions/node/v6.2.0/lib/node_modules/npm/lib/outdated.js:62:20
11 error typeerror     at /Users/samh/.nvm/versions/node/v6.2.0/lib/node_modules/npm/node_modules/read-package-tree/rpt.js:186:9
11 error typeerror     at f (/Users/samh/.nvm/versions/node/v6.2.0/lib/node_modules/npm/node_modules/once/once.js:17:25)
11 error typeerror     at /Users/samh/.nvm/versions/node/v6.2.0/lib/node_modules/npm/node_modules/read-package-tree/rpt.js:152:20
11 error typeerror     at f (/Users/samh/.nvm/versions/node/v6.2.0/lib/node_modules/npm/node_modules/once/once.js:17:25)
11 error typeerror     at then (/Users/samh/.nvm/versions/node/v6.2.0/lib/node_modules/npm/node_modules/read-package-tree/rpt.js:121:22)
11 error typeerror     at /Users/samh/.nvm/versions/node/v6.2.0/lib/node_modules/npm/node_modules/read-package-tree/rpt.js:81:20
11 error typeerror     at FSReqWrap.oncomplete (fs.js:117:15)
12 error typeerror This is an error with npm itself. Please report this error at:
12 error typeerror     <http://github.com/npm/npm/issues>
13 verbose exit [ 1, true ]
```
